.github/workflows/wash.yml

```diff
@@ -118,17 +117,35 @@ jobs:
         run: make test-integration-ci
         working-directory: ./crates/wash-cli
 
+  build_current:
+    name: Build `wash` for e2e tests
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+      - run: rustup show
+      - uses: Swatinem/rust-cache@82a92a6e8fbeee089604da2575dc567ae9ddeaab
+        with:
+          shared-key: 'ubuntu-22.04-shared-cache'
+      - name: Build wash
+        run: cargo build
+        working-directory: ./crates/wash-cli
+      - name: Upload wash binary
+        uses: actions/upload-artifact@6f51ac03b9356f520e9adb1b1b7802705f340c2b
+        with:
+          name: wash
+          path: ./target/debug/wash
```

Comment:
We might want to move this to an action since were doing in the `examples-*` workflows as well. Those ones use [`taiki-e/cache-cargo-install-action`](https://github.com/taiki-e/cache-cargo-install-action) which means less steps to maintain and same outcome:

https://github.com/wasmCloud/wasmCloud/blob/d44138d7937df8e08b0b3c768441ae2f90332a9a/.github/workflows/examples-components.yml#L81-L87

Reply:
Oh I didn't realize it could work that way. `install-action` picks up on this properly in the `./.github/actions/wasm-lang-toolchain-setup` then?

Reply:
Yeah, the cache-cargo-install-action will set the cache key against a commit meaning it should only ever build once for a given commit.

I just added #3896 an action which makes this more re-usable.

Reply:
It's merged! I'm going to make another comment